include/picongpu/plugins/binning/axis/LogAxis.hpp

```diff
@@ -0,0 +1,287 @@
+                                if(logVal < logMax)
+                                {
+                                    /**
+                                     * Precision errors?
+                                     * Is the math floor necessary?
+                                     */
+                                    binIdx = math::floor(((logVal - logMin) * scaling) + 1);
+                                    if(!overflowEnabled)
+                                    {
+                                        enableBinning = true;
+                                        binIdx = binIdx - 1;
+                                    }
+                                }
+                                else
+                                    binIdx = nBins - 1;
+                            }
+                        }
+
+                        validIdx = validIdx && enableBinning;
+                        return binIdx;
+                    }
+                };
+
+                LogAxisKernel lAK;
+
+                struct BinWidthKernel
+                {
+                    typename pmacc::HostDeviceBuffer<T, 1>::DBuffer::DataBoxType widthsDeviceBox;
+
+                    BinWidthKernel(pmacc::HostDeviceBuffer<T, 1>& widths)
+                        : widthsDeviceBox{widths.getDeviceBuffer().getDataBox()}
+                    {
+                    }
+
+                    ALPAKA_FN_ACC T getBinWidth(uint32_t idx) const
+                    {
+                        return widthsDeviceBox(idx);
+                    }
+                };
+
+                LogAxis(
+                    AxisSplitting<T_Attribute> axSplit,
+                    T_AttrFunctor attrFunctor,
+                    std::string label,
+                    std::array<double, 7> units) // add type T to the default label string
+                    : axisSplit{axSplit}
+                    , label{label}
+                    , units{units}
+                    , lAK{attrFunctor, axisSplit.enableOverflowBins}
+                {
+                    initLAK();
+
+                    binWidths.reserve(axisSplit.nBins);
+                    // Calculate the logarithmic spacing factor
+                    T factor = std::pow(2., (lAK.logMax - lAK.logMin) / axisSplit.nBins);
+                    // Calculate bin widths
+                    T edge = axisSplit.m_range.min;
+                    for(int i = 0; i < axisSplit.nBins; ++i)
+                    {
+                        T next_edge = factor * edge;
+                        binWidths.emplace_back(next_edge - edge);
+                        edge = next_edge;
+                    }
+                }
+
+                void initLAK()
+                {
+                    // do conversion to PIC units here, if not auto
+                    // auto picRange = toPICUnits(axSplit.range, units);
+                    auto min = toPICUnits(axisSplit.m_range.min, units);
+                    PMACC_ASSERT(0. < min);
```

Comment:
Maybe `PMACC_VERIFY` is better because than it will always be checked.

Reply:
done